runelite-client/src/main/java/net/runelite/client/ui/overlay/OverlayRenderer.java

```diff
@@ -381,18 +382,22 @@ private void safeRender(Client client, Overlay overlay, OverlayLayer layer, Grap
 
 		final OverlayPosition position = overlay.getPosition();
 
-		// Set font based on configuration
-		if (position == OverlayPosition.DYNAMIC)
+		// Set font rendering properties like the OS's font rendering
+		Toolkit tk = Toolkit.getDefaultToolkit();
+		Map desktopHints = (Map)(tk.getDesktopProperty("awt.font.desktophints"));
+		if (desktopHints != null)
 		{
-			subGraphics.setFont(runeLiteConfig.fontType().getFont());
+			graphics.addRenderingHints(desktopHints);
 		}
-		else if (position == OverlayPosition.TOOLTIP)
+
+		// Set font based on configuration
+		if (position == OverlayPosition.TOOLTIP)
 		{
-			subGraphics.setFont(runeLiteConfig.tooltipFontType().getFont());
+			subGraphics.setFont(runeLiteConfig.tooltipFont());
 		}
 		else
 		{
-			subGraphics.setFont(FontManager.getRunescapeFont());
+			subGraphics.setFont(runeLiteConfig.overlayFont());
```

Comment:
Why was support for changing font for dynamic overlays separately removed? It should be put back

Reply:
Probably, as you are removing the possibility to choose smaller font for tooltips than for panels with this change, what was one of the issues before why the option for dynamic overlays was added.